uber/menu.py

```diff
@@ -0,0 +1,51 @@
+from uber.common import *
+
+
+class MenuItem:
+    access = None   # list of permission levels allowed to display this menu
+    href = None     # link to render
+    submenu = None  # submenu to show
+    name = None     # name of Menu item to show
+
+    def __init__(self, href=None, access=None, submenu=None, name=None):
+        if submenu:
+            self.submenu = listify(submenu)
+        else:
+            self.href = href
+
+        self.name = name
+        self.access = access
+
+    def __getitem__(self, key):
+        for sm in self.submenu:
+            if sm.name == key:
+                return sm
+
+    def append_menu_item(self, m):
+        # if we aren't a submenu, convert us to one now
+        if not self.submenu and self.href:
+            self.submenu = [MenuItem(name=self.name, href=self.href)]
+            self.href = None
+
+        self.submenu.append(m)
+
+
+c.MENU = MenuItem(name='Root', submenu=[
```

Comment:
btw, It may not be necessary anymore to have a 'Root' Menu object like that, i -think- it was just a convenience for my JSON exporter to only have to look at one place.  it does make some sense that you would only interact with the main menu by using the methods in Root's Menu, just for the sake of consistency.